dot/core/service.go

```
@@ -127,40 +127,68 @@ func (s *Service) StorageRoot() (common.Hash, error) {
 
 // HandleBlockImport handles a block that was imported via the network
 func (s *Service) HandleBlockImport(block *types.Block, state *rtstorage.TrieState) error {
-	return s.handleBlock(block, state)
+	err := s.handleBlock(block, state)
+	if err != nil {
+		return fmt.Errorf("handling block: %s", err)
+	}
+
+	isBestBlock := false
+	bestBlockHash := s.blockState.BestBlockHash()
+
+	if bestBlockHash.Equal(block.Header.Hash()) {
+		isBestBlock = true
+	}
+
+	blockAnnounce, err := createBlockAnnounce(block, isBestBlock)
+	if err != nil {
+		logger.Errorf("creating block announce: %s", err)
+	}
+
+	s.net.GossipMessage(blockAnnounce)
+	return nil
 }
 
 // HandleBlockProduced handles a block that was produced by us
 // It is handled the same as an imported block in terms of state updates; the only difference
 // is we send a BlockAnnounceMessage to our peers.
 func (s *Service) HandleBlockProduced(block *types.Block, state *rtstorage.TrieState) error {
-	if err := s.handleBlock(block, state); err != nil {
-		return err
+	err := s.handleBlock(block, state)
+	if err != nil {
+		return fmt.Errorf("handling block: %s", err)
 	}
 
+	const isBestBlock = true
+	blockAnnounce, err := createBlockAnnounce(block, isBestBlock)
+	if err != nil {
+		logger.Errorf("creating block announce: %s", err)
```

why not return the error?